search/src/Components/upload/uploader_zip.js

```diff
@@ -1,61 +1,131 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+
 import 'react-dropzone-uploader/dist/styles.css';
 import Dropzone from 'react-dropzone-uploader';
 import Cookies from 'js-cookie';
 
+import axios from 'axios';
+
+import Uppy from '@uppy/core';
+import '@uppy/core/dist/style.css';
+import '@uppy/dashboard/dist/style.css';
+import Tus from '@uppy/tus';
+import { Dashboard, useUppy } from '@uppy/react';
+
+import {jsonSchemaTitle} from '../error/utils';
+
+const baseURL = new URL(window.location.origin);
+
+
+const TUS_ENDPOINT = baseURL + 'tus/files/';
+const TUS_CHUNK_SIZE = 1024 * 1024 * 10;
 
-const UploaderZip  = props => {
-    const baseURL = new URL(window.location.origin);
-    const getUploadParams = ({ file, meta }) => {
-        const body = new FormData()
-        body.append('upload_id', props.upload_id)
-        body.append('images', props.images)
-        body.append('upload_image_zip', file)
-        return { url: baseURL + 'api/upload_image_zip/',
-                 mode: 'same-origin',
-                 headers: {'X-CSRFToken': Cookies.get('csrftoken')},
-                 body
-               }
+
+function getTusUploadFile(file) {
+    const url = file.tus.uploadUrl;
+    if( url ) {
+        const m = url.match(RegExp('^' + TUS_ENDPOINT + '(.*)$'));
+        if( m ) {
+            return m[1];
+        }
     }
-  
-    const handleChangeStatus = ({ meta, file, xhr }, status) => {
-        if (status === 'done'){
-            const res = JSON.parse(xhr.response)
-            if (res.upload_id === props.upload_id) {
-                if (res.missing_images.length === 0) {
-                    props.handleValidation(true)
-                    props.handleErrorMessage("")
+    return null;
+}
+
+
+class UploaderUppyZip extends React.Component {
+
+    constructor(props) {
+        super(props);
+        
+        this.uppy = new Uppy({
+            id: 'zipfiles',
+            restrictions: {
+                maxNumberOfFiles: 1,
+                minNumberOfFiles: 1,
+                allowedFileTypes: [ "application/zip" ],
+            },
+            autoProceed: true,
+        }).use(Tus, {
+            endpoint: TUS_ENDPOINT,
+            withCredentials: true,
+            headers: {'X-CSRFToken': Cookies.get('csrftoken')},
+            chunkSize: TUS_CHUNK_SIZE
+        });
+    }
+
+    componentDidMount() {
+
+        this.uppy.on("complete", (result) => {
+            const baseURL = new URL(window.location.origin);
+            const files = result.successful;
+            if( files.length === 1 ) {
```

Comment:
I find that the function below gets very deep because of this structure. Do you not prefer:
```suggestion
            if( files.length !== 1 ) {
                console.log("Got wrong number of successful uploaded files");
                this.props.handleValidation(false);
                this.props.handleErrorMessage("Upload zipfile failed");
                return;
              }
```
I find it improves coherence and keeps the core implementation not so nested.

Reply:
Yes, good point. I've done this and also made the other error condition (failed parse from the uppy URL) work the same way